xgcm/grid.py

```diff
@@ -555,6 +569,15 @@ def cumsum(self, da, axis, **kwargs):
         return ax.cumsum(da, **kwargs)
 
 
+def add_slice(da, dim, sl, added):
+    axis = da.get_axis_num(dim)
+    t_order = list(range(len(da.data.shape)))
+    t_order[axis] = 0
+    t_order[0] = axis
+    data = np.transpose(da.data.copy(), axes=t_order)
```

Comment:
Does calling transpose on dask data trigger computation? If so, this is forbidden.

Reply:
Can you explain what this function does?

Reply:
This function was my attempt to enable the 'wrapping' of values around a discontinuity. The basic idea is to add a 'wrap value' to the first or last (depending on the shift_to direction) slice from an arbitrary axis (xarray dimension). Is there a way to selectively replace values in a dask array without triggering computation?

Alternatively we might have to compute and then return to a dask array?